designer/report-designer/src/main/java/org/pentaho/reporting/designer/core/actions/elements/MorphAction.java

```diff
@@ -106,6 +106,10 @@ public void actionPerformed( final ActionEvent e ) {
       if ( ObjectUtilities.equal( visualElement.getElementTypeName(), targetElementType.getMetaData().getName() ) ) {
         continue;
       }
+      if ( targetMetaData.getReportElementType().compareTo( ElementMetaData.TypeClassification.SUBREPORT ) == 0
```

Comment:
The lords of Java in their infinite wisdom invented Equals to test for object equality. Use it, they will be grateful!